kratos/geometries/coupling_geometry.h

```diff
@@ -337,6 +304,236 @@ template<class TPointType> class CouplingGeometry
+                        << "Projection of intersection spans failed. Global Coordinates on slave: "
+                        << global_coords << ", and global coordinates on master: "
+                        << global_coords_master << ". Difference: " << norm_2(global_coords - global_coords_master)
+                        << " larger than model tolerance: " << model_tolerance << std::endl;
+
+                    // If success == 0, it is considered that the projection is on one of the boundaries.
+                    intersection_master_spans_on_master_spans.push_back(local_coords_master[0]);
+                }
+            }
+            std::vector<double> all_intersections;
+            MergeSpans(all_intersections, intersection_master_spans, intersection_master_spans_on_master_spans);
+
+            IntegrationPointUtilities::CreateIntegrationPoints1D(rIntegrationPoints, all_intersections, rIntegrationInfo.GetNumberOfIntegrationPointsPerSpan(0));
+        }
+    }
+
+    /* @brief This method creates a list of quadrature point geometries
+     *        from a list of integration points.
+     *        Initially, all integration points are transformed to
+     *        quadrature point geometries on the master. Then all points
+     *        are mapped to all slave geometries. Finally, coupling geometries,
+     *        containing quadrature point geometries per integration point on
+     *        master and all slaves are created.
+     *
+     * @param rResultGeometries list of quadrature point geometries.
+     * @param NumberOfShapeFunctionDerivatives the number of evaluated
+     *        derivatives of shape functions at the quadrature point geometries.
+     *
+     * @see quadrature_point_geometry.h
+     */
+    void CreateQuadraturePointGeometries(
+        GeometriesArrayType& rResultGeometries,
+        IndexType NumberOfShapeFunctionDerivatives,
+        const IntegrationPointsArrayType& rIntegrationPoints,
+        IntegrationInfo& rIntegrationInfo) override
+    {
+        const double model_tolerance = 1e-3;
+
+        const SizeType num_integration_points = rIntegrationPoints.size();
+
+        if (rResultGeometries.size() != num_integration_points) {
+            rResultGeometries.resize(num_integration_points);
+        }
+
+        // Create quadrature points on master
+        GeometriesArrayType master_quadrature_points(num_integration_points);
+        mpGeometries[0]->CreateQuadraturePointGeometries(
+            master_quadrature_points,
+            NumberOfShapeFunctionDerivatives,
+            rIntegrationPoints,
+            rIntegrationInfo);
+
+        // Compute vector of location
+        CoordinatesArrayVectorType integration_points_global_coords_vector(num_integration_points);
+        for (SizeType i = 0; i < num_integration_points; ++i) {
+            integration_points_global_coords_vector[i] = master_quadrature_points[i].Center();
```

Comment:
Do you mean the center of the geometry or the coordinates of the quadrature point?

Reply:
The center of a quadrature point does provide the global location of it. Yet, it is the most efficient way to access its location.